src/RotationMatrix.java

```diff
@@ -0,0 +1,184 @@
+    }
+
+    /**
+     * Constructor for basic rotation around Y axis.
+     *
+     * @param angle Angle in degrees of counterclockwise rotation around Y axis.
+     */
+    public static RotationMatrix basicRotationY(double angle) {
+        return new RotationMatrix(angle, new Vector3D(0, 1, 0));
+    }
+
+    /**
+     * Constructor for basic rotation around Z axis.
+     *
+     * @param angle Angle in degrees of counterclockwise rotation around Z axis.
+     */
+    public static RotationMatrix basicRotationZ(double angle) {
+        return new RotationMatrix(angle, new Vector3D(0, 0, 1));
+    }
+
+
+    /**
+     * Method to rotate a vector.
+     *
+     * @param vector Vector3D to rotate.
+     * @return Result of the rotation.
+     */
+    public Vector3D rotate(Vector3D vector) {
+        double[] result = new double[]{0, 0, 0};
+
+        for (int i = 0; i < SIZE; i++) {
+            for (int j = 0; j < SIZE; j++) {
+                result[i] += this.getElement(i, j) * vector.getElement(j);
+            }
+        }
+
+        return new Vector3D(result);
+    }
+
+    /**
+     * Method to assert if to Rotation Matrices are equal.
+     *
+     * @return Returns True if they are equal, and False otherwise.
+     */
+    public boolean equals(RotationMatrix compared) {
+        for (int i = 0; i < SIZE; i++) {
+            for (int j = 0; j < SIZE; j++) {
+                if (this.getElement(i, j) != compared.getElement(i, j)) {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Method to assert if to Rotation Matrices are equal.
+     *
+     * @return Returns True if they are equal, and False otherwise.
+     */
+    @Override
+    public boolean equals(Object compared) {
+        return compared instanceof RotationMatrix && this.equals((RotationMatrix) compared);
+    }
+
+
+    /**
+     * Method to convert a RotationMatrix to a string.
+     *
+     * @return Returns the RotationMatrix in the form ( a b c ; d e f ; g h i ).
+     */
+    @Override
+    public String toString() {
+
+        StringBuilder result = new StringBuilder();
+
+        result.append("( ");
+        for (int i = 0; i < SIZE; i++) {
+            for (int j = 0; j < SIZE; j++) {
+                result.append(this.getElement(i, j) + " ");
+            }
+            if (i == SIZE - 1) {
+                result.append(" )");
+            } else {
+                result.append("; ");
+            }
+        }
+
+        return result.toString();
+    }
+
+    //************ Getters & Setters ************//
+
+    /**
+     * Method to get the element i,j of the matrix.
+     *
+     * @return Returns the element in the ith row and jth column of the matrix.
+     */
+    public double getElement(int i, int j) {
+        if (i < 0 || i >= SIZE || j < 0 || j >= SIZE) {
+            throw new IllegalArgumentException("Attempting to access an inexistent element");
```

Comment:
This time the exception is right, but maybe we could improve the error message. How about adding the `i` and `j` to the  message and stating the matrix size (e.g. "Element out of bounds (4,3). Matrix size is 3x3")?.